### ISSUE TITLE
spring cloud gateway transaction is empty

### ISSUE THREAD

**Issue by 295black** [elastic/apm-agent-java#1934]
We kindly ask to post all questions and issues on the [Discuss forum](https://discuss.elastic.co/c/apm) first. In addition to awesome, knowledgeable community contributors, core APM developers are on the forums every single day to help you out as well, so your questions will reach a wider audience there. If we confirm that there is a bug, then you can open an issue in this repo.

## Describe the bug

I use gateway 3.0.3,apm-agent 1.24.0 . In the apm-server,service is exist,but transaction is empty.
![image](https://user-images.githubusercontent.com/12394342/127103596-cbd0894e-336c-491b-aeb7-c4618bd88e9c.png)

![image](https://user-images.githubusercontent.com/12394342/127103185-4b6fe1a0-fd07-4336-81ea-3c409143e310.png)
![image](https://user-images.githubusercontent.com/12394342/127103282-c90adac8-7f9d-4406-bee3-a48377035e0a.png)
![image](https://user-images.githubusercontent.com/12394342/127103367-95c4e860-2420-4d25-85c1-353ed770efd3.png)

## Debug logs

<details>
<summary>Click to expand</summary>

```
replace this line with your debug logs
```
</details>

**Comment by eyalkoren** [comment]
You should be aware of #1924 , which refers to a snapshot that you would need to try instead of 1.24.0. This snapshot contains WebFlux support for `use_path_as_transaction_name`, which is not yet merged.

@dhannappe can you provide advice here? Do you have any idea why @295black doesn't get any transaction at all?

**Comment by dhannappe** [comment]
For version 1.24.0
Parameter should be 
disable_instrumentations
With no value
Doc below 
https://www.elastic.co/guide/en/apm/agent/java/current/config-core.html#config-disable-instrumentations

**Comment by abdulazizali77** [comment]
@295black i do not know whether this is related to your problem, but whilst developing the webflux client feature, i faced issues where the `WeakConcurrentHash` couldnt retrieve the saved object/Span/Transaction, i didnt get to the bottom of it yet, but i 'fixed' my problem by using a simple `ConcurrentHash` instead of the prevalent `WeakConcurrentHash` (which may introduce gc/memory problems later)

**Comment by 295black** [comment]
WeakHashMap? I never use it,@abdulazizali77 。thanks for your help, my gateway is very clear, but still have this problem.

**Comment by 295black** [comment]
when i use resilience4j.circuitbreaker, transaction will be empty. @eyalkoren

**Comment by eyalkoren** [comment]
@295black did you change your configuration to `disable_instrumentations=` instead of `enable_experimental_instrumentations=false` like @dhannappe proposed? 
You can also leave the same configuration and upgrade the agent to 1.25.0. 
See [release notes](https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-1.x.html#release-notes-1.25.0).

**Comment by 295black** [comment]
I use 1.25.0 agent, How to set configuration?
![image](https://user-images.githubusercontent.com/12394342/127301318-9a0e67e1-4626-47d3-9127-5f4fbfa26595.png)
i still got POST unknown route.

**Comment by eyalkoren** [comment]
OK, that's better. 

Now try [this snapshot](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-java%2Fapm-agent-java-mbp/detail/PR-1906/26/artifacts), where this issue is fixed, as described in #1924 .

This fix will be included in the next release.

**Comment by 295black** [comment]
this snapshot is ok! thanks.
but i have a new problem.
i use 
```
<dependency>
	<groupId>org.apache.httpcomponents</groupId>
	<artifactId>httpclient</artifactId>
	<version>4.5.13</version>
</dependency>
```
can not trace the http request.
